src/main/scala/sigmastate/Values.scala

```diff
@@ -424,6 +424,16 @@ object Values {
       case _ =>
         illegalArgs("ConcreteCollection", "(IndexedSeq, SType)", cs)
     }
+
+    override def equals(obj: scala.Any): Boolean = obj match {
+      case cc: ConcreteCollection[_] =>
```

Comment:
Please check if we can simple move elementType to regular args. We already have ConcreteCollection.apply with implicit arg.

Reply:
Oops, I missed it. Thanks!